ext/pgsql/pgsql.c

```diff
@@ -6248,3 +6246,29 @@ PHP_FUNCTION(pg_socket_poll)
 
 	RETURN_LONG((zend_long)PQsocketPoll(socket, (int)read, (int)write, (int)ts));
 }
+
+#if defined(HAVE_PG_SET_CHUNKED_ROWS_MODE)
+PHP_FUNCTION(pg_set_chunked_rows_mode)
```

Comment:
The name of the function confuses me, is this setting a "mode" or the number of rows to chunk?

Reply:
"mode" refers to chunk mode vs single row mode (which is another libpq call, thus the exposition of PGSQL_TUPLES_CHUNK constant) but yes it set the number of rows.

Reply:
Does it possibly make sense to name the function we expose to userland in a slightly more sensible way? I think just replacing `mode` with `size` would possibly do.

And thus it mean, it can only fail if trying to set up a chunk size for results that cannot be chunked?

Reply:
can do indeed.